File: s4s-backend/src/main/java/com/s4s/database/model/Rating.java
package com.s4s.database.model;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class Rating {
    private String uid;

    private String rating;
    private String name;
    private String comment;

    private List<String> imageNames = new ArrayList<>();
    private List<String> imageUrl = new ArrayList<>();

    private String creator;
    private String creatorNickName;
    private Date updatedAt;
    private String createdAtString;

    private String sightId;
    public String sightName;

    public String getSightId(){return  sightId;}

    public String getUid() {
        return uid;
    }

    public void setUid(String uid) {
        this.uid = uid;
    }

    public String getRating() {
        return rating;
    }

    public void setRating(String rating) {
        this.rating = rating;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getComment() {
        return comment;
    }

    public void setComment(String comment) {
        this.comment = comment;
    }

    public String getCreator() {
        return creator;
    }

    public void setCreator(String creator) {
        this.creator = creator;
    }

    public Date getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Date updatedAt) {
        this.updatedAt = updatedAt;
    }

    public List<String> getImageNames() {
        return imageNames;
    }

    public void setImageNames(List<String> imageNames) {
        this.imageNames = imageNames;
    }

    public void setSightId(String sightId) {
        this.sightId = sightId;
    }

    public String getCreatorNickName() {
        return creatorNickName;
    }

    public void setCreatorNickName(String creatorNickName) {
        this.creatorNickName = creatorNickName;
    }

    public List<String> getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(List<String> imageUrl) {
        this.imageUrl = imageUrl;
    }

    public String getCreatedAtString() {
        return createdAtString;
    }

    public void setCreatedAtString(String createdAtString) {
        this.createdAtString = createdAtString;
    }
}

File: s4s-frontend/src/model/rating.ts
/**
 * Model for a rating
 * Component written by Michael Fahrafellner
 * creation date: 16.10.2021
 * last change done by: Michael Fahrafellner
 */
export class Rating {
    uid: string;
    rating: number;
    comment: string = "";
    imageNames: string[] = []
    createdAt:string;
    creator:String;
    updatedAt:string;
    imageUrl: string[] = []
    sightId: string;
    sightName: string;

    public InitRating(rating:any){
        this.uid = rating.uid;
        this.rating = rating.rating;
        this.comment = rating.comment;
        this.sightId = rating.sightId;
        this.sightName = rating.sightName;
        var images = [];
        rating.imageNames.forEach(rat => {
            images.push(rat)
        })
        this.imageNames = images;

        this.creator = rating.creatorNickName;

        if(rating.createdAt!=undefined){
            this.createdAt = rating.createdAtString;
        }
    }
}